imap/jmap_mail.c

```diff
@@ -3278,6 +3297,40 @@ static int guidsearch_expr_eval(struct conversations_state *cstate,
                 conversation_fini(&conv);
                 return ret;
             }
+        case GSEOP_UNFUZZ:
+            {
+                /* Re-evaluate the match againt actual message data */
+                char email_id[JMAP_EMAILID_SIZE];
+                struct message_guid guid;
+                struct mailbox *mbox = NULL;
+                char *mboxname = NULL;
+                msgrecord_t *mr = NULL;
+                message_t *msg = NULL;
+                uint32_t uid;
+                int ret = 0;
+
+                message_guid_decode(&guid, match->guidrep);
+                jmap_set_emailid(&guid, email_id);
+
+                if (!jmap_email_find(e->req, email_id, &mboxname, &uid) &&
```

Comment:
This is a significant extra expense to these queries.

Reply:
Yes, but I see no way around it with the current indexing.  Maybe limiting this to only when the search string contains '@' but if we're going to re-index then it doesn't matter.

Is there anyway to implement prefix, suffix, and complete string matches via Xapian?  If not, we're stuck doing the above for those matches.